vendor/github.com/ethereum/go-ethereum/eth/downloader/downloader.go

```diff
@@ -405,8 +404,12 @@ func (d *Downloader) synchronise(id string, hash common.Hash, td *big.Int, mode
 // syncWithPeer starts a block synchronization based on the hash chain from the
 // specified peer and head hash.
 func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.Int) (err error) {
```

Comment:
why here?

Reply:
@dshulyak this is a previous race condition I fixed last week, same problem.